Form/DataTransformer/ModelsToArrayTransformer.php

```diff
@@ -39,20 +39,46 @@ class ModelsToArrayTransformer implements DataTransformerInterface
 
     /**
      * @var ModelChoiceList
+     * @deprecated
      */
     protected $choiceList;
 
     /**
      * ModelsToArrayTransformer constructor.
      *
      * @param ModelChoiceList|LazyChoiceList|ModelChoiceLoader $choiceList
-     * @param ModelManagerInterface                            $modelManager
+     * @param $modelManager
      * @param $class
      *
      * @throws RuntimeException
      */
-    public function __construct($choiceList, ModelManagerInterface $modelManager, $class)
+    public function __construct($choiceList, $modelManager, $class)
+    {   
+        /*
+        NEXT MAJOR: Remove condition , magic methods, legacyConstructor() method, $choiceList property and argument
+        __construct() signature should be : public function __construct(ModelManager $modelManager, $class)
+         */
+        
+        $args = func_get_args(__FUNCTION__);
```

Comment:
As you want… do based upon what you think is easier to understand, speed does not matter here.